README.md

```diff
@@ -1,14 +1,615 @@
+
+[//]: # ()
+[//]: # (For Groovy:)
+
+[//]: # ()
+[//]: # (```groovy)
+
+[//]: # (implementation 'com.ably.chat:android')
+
+[//]: # (```)
+
+[//]: # ()
+[//]: # (For Kotlin Script &#40;`build.gradle.kts`&#41;:)
+
+[//]: # ()
+[//]: # (```kotlin)
+
+[//]: # (implementation&#40;"com.ably.chat:android"&#41;)
+
+[//]: # (```)
+
+[//]: # ()
+[//]: # (### Dependency on ably-android)
+
+[//]: # ()
+[//]: # (Certain functionalities are powered by the [ably-android]&#40;https://github.com/ably/ably-java&#41; library. )
+
+[//]: # (The `ably-android` library is included as an api dependency within the Chat SDK, so there is no need to manually add it to your project.)
+
+## Versioning
+
+The Ably client library follows [Semantic Versioning](http://semver.org/). See https://github.com/ably/ably-chat-kotlin/tags for a list of
+tagged releases.
+
+## Instantiation and authentication
+
+To instantiate the Chat SDK, create an [Ably client](https://ably.com/docs/getting-started/setup) and pass it into the
+Chat constructor:
+
+```kotlin
+import com.ably.chat.ChatClient
+import io.ably.lib.realtime.AblyRealtime
+import io.ably.lib.types.ClientOptions
+
+val realtimeClient = AblyRealtime(
+    ClientOptions().apply {
+        key = "<api-key>"
+        clientId = "<client-id>"
+    },
+)
+
+val chatClient = ChatClient(realtimeClient)
+```
+
+You can use [basic authentication](https://ably.com/docs/auth/basic) i.e. the API Key directly for testing purposes,
+however it is strongly recommended that you use [token authentication](https://ably.com/docs/auth/token) in production
+environments.
+
+To use Chat you must also set a [`clientId`](https://ably.com/docs/auth/identified-clients) so that clients are
+identifiable. If you are prototyping, you can use `java.util.UUID` to generate an ID.
+
+## Connections
+
+The Chat SDK uses a single connection to Ably, which is exposed via the `ChatClient#connection` property. You can use this
+property to observe the connection state and take action accordingly.
+
+### Current connection status
+
+You can view the current connection status at any time:
+
+```kotlin
+val connectionStatus = chat.connection.status
+val connectionError = chat.connection.error
+```
+
+### Subscribing to connection status changes
+
+You can subscribe to connection status changes by registering a listener, like so:
+
+```kotlin
+val subscription = chat.connection.onStatusChange { statusChange: ConnectionStatusChange ->
+    println(statusChange.toString())
+}
+```
+
+To stop listening to changes, `unsubscribe` method on returned subscription instance:
+
+```kotlin
+subscription.unsubscribe()
+```
+
+## Chat rooms
+
+### Creating or retrieving a chat room
+
+You can create or retrieve a chat room with name `"basketball-stream"` this way:
+
+```kotlin
+val room = chat.rooms.get("basketball-stream", RoomOptions(reactions = RoomReactionsOptions()))
```

Comment:
Would it be worth an example with, say, a custom typing timeout, so the customisability is shown?

Reply:
done